rfcs/0000-api-cr-crowding.md

```diff
@@ -0,0 +1,337 @@
+[occupancies.txt](https://github.com/google/transit/pull/240/files), which contains information
+about "usual or expected in-vehicle occupancy levels". The underlying data can apply to either a
+whole trip or individual stop times along the trip. A given occupancy specifies its
+`occupancy_status` which is an enum ranging from `0` for "Empty" to `6` for "Full". Similar to how
+we represent vehicles in the API, the API response will be the enum word rather than the number.
+Our feed will have an _extra_ field, currently not part of the spec, but which is being advocated
+for, `occupancy_percentage`, which is a value from `0.0` to `1.0`. Each occupancy has some
+validity constraints, namely the start and end dates and days of the week for which they're valid.
+A trip could therefore have several related occupancies.
+
+Our Keolis source data is at the trip level, and that is the only producer for which we will have
+static occupancy data in the near term. Initially, we will put this data directly in the API and
+not in our GTFS feed since the data updates as frequently as once/hour, and we don't want to
+update our GTFS static feed that often. However, we should ensure the API interface works _as if_
+it were coming from the GTFS feed, in case we do so in the future. While we're not given this
+data, we can arbitrarily say that the occupancies are valid for two weeks and apply to all days of
+the week. A weekend train trip generally doesn't run on weekdays, but if it were to (e.g. Labor
+Day running Sunday service), we'd want the predicted occupancy to be considered valid.
+
+Our API largely matches the GTFS feed and is simply a convenient way of accessing that data. While
+there are a few GTFS resources that we use in the API that are not exposed directly (e.g.
+transfers, calendar) and one major one that's named differently (schedule vs stoptime), the
+default expectation is that GTFS resources are available in the API.
+
+To that end, the API has a new `/occupancies` endpoint for an `Occupancy` resource. While the
+occupancy must have an ID – which we'll construct as `occupancy-[trip-id]` – there's no
+`/occupancies/[id]` endpoint, akin to `/predictions` or `/schedules`.
+
+For now, due to the experimental and malleable nature of the proposal, we want to start small and
+limit the coupling of occupancies with the rest of our code. So at a minimum we will provide just
+a single `/occupancies` endpoint that returns all current occupancies, with their trip
+relationship, like such:
+
+```
+GET /occupancies
+
+{
+  data: [
+    {
+      "attributes": {
+        "occupancy_status": MANY_SEATS_AVAILABLE,
+        "occupancy_percentage": 0.11,
+        "start_date": [today],
+        "end_date": [2 weeks from today],
+        "monday": 1,
+        "tuesday": 1,
+        "wednesday": 1,
+        "thursday": 1,
+        "friday": 1,
+        "saturday": 1,
+        "sunday": 1
+      }
+      "id": occupancy-[trip-id]-all,
```

Comment:
This ID doesn't support having the same trip with multiple occupancies (say for example if there's an updated occupancy estimate for the future but not today. Maybe include the `start_date`?

Reply:
I thought about that. But *technically* even including the `start_date` isn't enough since we could use a scheme with the same start/end dates, and different valid days, to, e.g., distinguish weekday and weekend travel. I'm not sure how to unequivocally get a unique ID for all possible kinds of the data, short of just repeating all the data in the ID! (or hashing it).

But I don't think clients will really rely that much on the continuity of the ID, so I figured it would be fine to just use something that works now with the data we've got, and change the shape later, if need be.